scripts/test-aws-creds.md

```diff
@@ -0,0 +1,85 @@
+### Introduction
+This documentation talks about how AWS `credentials` are managed for running
+ACK service controller `end-to-end` tests locally and on `Prow`.
+
+The actual bash scripts to understand this functionality can be confusing so
+this document serves as reference for debugging and making future updates for
+this functionality.
+
+### Overview
+While running `end-to-end` tests, AWS credentials are required at following two places:
+1. Inside `Service Controller` which is being tested. These AWS credentials are used
+to invoke AWS APIs for managing AWS resources from ACK controller.
+2. Inside `Test Container` executing the `run-tests.sh` script. These AWS credentials
+are used for `bootstraping` and `cleaning` e2e resources, and also for calling AWS APIs
+in assertion statements that validate K8s resource state with actual AWS resource.
+
+### Service Controller
+For providing AWS credentials inside the service controller, the `kind-build-test.sh`
+script assumes an IAM role. For `Prow job`, the script assumes the test role provided
+by AWS service teams. This role name is stored in `SSM parameter store`. For local
+testing, the script assumes the role name from environment variable `ACK_ROLE_ARN`
+
+For both `Prow` and `local testing`, a background thread is started, which
+refreshes AWS credentials every `50 minutes`. Then updates the AWS credentials
+environment variables in service controller `k8s deployment` and restarts the
+`k8s deployment`. This `background thread` and `k8s deployment restart` is required
+for tests which run longer than `60 minutes` to avoid AWS credentials expiration.
+
+A good followup question for above background refresh will be **"Why didn't
+you increase the ASSUME_ROLE_SESSION_DURATION to maximum allowed 12 hours?
```

Comment:
`ASSUME_ROLE_SESSION_DURATION` refers to an environment variable and therefore actually *should* be enclosed in backticks :)